### PR TITLE
minissl read_nonblock fix

### PR DESCRIPTION
**Pull request by MSP-Greg**
Travis is having intermittent errors across various releases.  This seems to fix issue.  Just went thru a bit of nonblock code variations in ruby core.

Please review, as I haven't really been outside of this code/file...

### CODE REVIEW

**Comment by MSP-Greg** [comment]
@nateberkopec

I hate it when this happens.

Travis:
My fork build [passed](https://travis-ci.org/MSP-Greg/puma/builds/290236638).  Travis here did two builds, first #2534 [passed](https://travis-ci.org/puma/puma/builds/290250048), second #2535 [failed](https://travis-ci.org/puma/puma/builds/290250330).

Appveyor:
My fork [passed](https://ci.appveyor.com/project/MSP-Greg/puma/build/1.0.7), Appveyor here (just one build) [failed](https://ci.appveyor.com/project/nateberkopec/puma/build/1.0.172) only on trunk, with a 'Could not build wrapper using Ragel' error on [line 114 of the log](https://ci.appveyor.com/project/nateberkopec/puma/build/1.0.172/job/3xyrh5dtvvbh67sr#L114).  I haven't seen that in previous tests.  I can add a ragel package to Appveyor, but it's not normally installed.  Odd that the error just appeared.

File: lib/puma/minissl.rb [hunk 1]
```diff
@@ -43,19 +48,20 @@ def read_nonblock(size, *_)
...
+              elsif data == :wait_readable
+                IO.select([@socket.to_io])
+              else
+                IO.select(nil, [@socket.to_io])
+              end
+            else
+              break
             end
-          end
+          end while true
```
Comment (olleolleolle) [inline lib/puma/minissl.rb]: Is there a specific reason not to use `loop do ... end` for this loop?

Comment (olleolleolle) [reply lib/puma/minissl.rb]: (It'd introduce a _block_, making `data` block-local.)

**Comment by MSP-Greg** [comment]
See #1464 and  7165775

**Pull request closed without merging**